utils/create-icons-map.js

```diff
@@ -0,0 +1,41 @@
+const fs = require('fs');
+const path = require('path');
+
+const svgFolder = '../assets/icons/';
+const iconMapFile = '../src/iconsMap.js';
+
+let iconsObject = '';
+
+let iconMapDir = path.dirname(iconMapFile);
+if (!fs.existsSync(iconMapDir)) {
+    fs.mkdirSync(iconMapDir);
+};
+
+
+
+fs.readdirSync(svgFolder).forEach(file => {
+    let fileName = file.split('.')[0];
+    let filePath = svgFolder + file;
+    if (fs.existsSync(filePath)) {
+        const iconBody = fs.readFileSync(filePath).toString();
+        const cleanedIcon = preparIcon(iconBody, fileName);
+        iconsObject += "'" + fileName + '\' : \'' + cleanedIcon + '\', \n';
+    }
+})
+
+const dataToWrite = 'export const Icons = {' + iconsObject + '};';
+
+fs.writeFileSync(iconMapFile, dataToWrite);
+
+console.log('Icons Map created!');
+
+
+function preparIcon(icon, iconName){
+    let iconLine = icon.replace(/\s{2,}/g, '');
```

Comment:
This will remove any 2 or more white spaces in a row.

Reply:
Exactly! I added here because want to make all the html from files in one line.

Reply:
This is bad practice. You assume that spaces between words would be exactly 1. And if I have XML like:
```
<tag1>
<tag2>
```
There's a single white space between those tags, and you'll not get a single line.

Reply:
Eli is correct, single new lines won't be caught. You could add \n to it like (\s{2,}|\n)